vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_util.cc

```diff
@@ -101,18 +105,126 @@ std::string GetSecondStepVerify() {
       id.c_str());
 }
 
-type::ExternalWalletPtr GetWallet(
-    std::map<std::string, type::ExternalWalletPtr> wallets) {
-  for (auto& wallet : wallets) {
-    if (wallet.first == constant::kWalletUphold) {
-      return std::move(wallet.second);
+type::UpholdWalletPtr GetWallet(LedgerImpl* ledger) {
+  DCHECK(ledger);
+  const std::string wallet_string = ledger->ledger_client()->GetStringState(
+      state::kWalletUphold);
+
+  if (wallet_string.empty()) {
+    return nullptr;
+  }
+
+  base::Optional<base::Value> value = base::JSONReader::Read(wallet_string);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Parsing of uphold wallet failed");
+    return nullptr;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    BLOG(0, "Parsing of uphold wallet failed");
+    return nullptr;
+  }
+
+  auto wallet = ledger::type::UpholdWallet::New();
+
+  auto* token = dictionary->FindStringKey("token");
+  if (token) {
+    wallet->token = *token;
+  }
+
+  auto* address = dictionary->FindStringKey("address");
+  if (address) {
+    wallet->address = *address;
+  }
+
+  auto* one_time_string = dictionary->FindStringKey("one_time_string");
+  if (one_time_string) {
+    wallet->one_time_string = *one_time_string;
+  }
+
+  auto status = dictionary->FindIntKey("status");
+  if (status) {
+    wallet->status = static_cast<ledger::type::WalletStatus>(*status);
+  }
+
+  auto* user_name = dictionary->FindStringKey("user_name");
+  if (user_name) {
+    wallet->user_name = *user_name;
+  }
+
+  auto* verify_url = dictionary->FindStringKey("verify_url");
+  if (verify_url) {
+    wallet->verify_url = *verify_url;
+  }
+
+  auto* add_url = dictionary->FindStringKey("add_url");
+  if (add_url) {
+    wallet->add_url = *add_url;
+  }
+
+  auto* withdraw_url = dictionary->FindStringKey("withdraw_url");
+  if (withdraw_url) {
+    wallet->withdraw_url = *withdraw_url;
+  }
+
+  auto* account_url = dictionary->FindStringKey("account_url");
+  if (account_url) {
+    wallet->account_url = *account_url;
+  }
+
+  auto* login_url = dictionary->FindStringKey("login_url");
+  if (login_url) {
+    wallet->login_url = *login_url;
+  }
+
+  auto* fees = dictionary->FindDictKey("fees");
+  if (fees) {
+    base::DictionaryValue* fees_dictionary;
+    if (fees->GetAsDictionary(&fees_dictionary)) {
+      for (base::DictionaryValue::Iterator it(*fees_dictionary);
```

Comment:
Nit: Can use an `auto` for the iterator to make this more concise.

Reply:
fixed

Reply:
No problem.